tracing/opentracing/http.go

```diff
@@ -0,0 +1,40 @@
+package opentracing
+
+import (
+	"net/http"
+
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context"
+)
+
+// ToHTTPRequest returns an http RequestFunc that injects an OpenTracing Span
+// found in `ctx` into the http headers. If no such Span can be found, the
+// RequestFunc is a noop.
+func ToHTTPRequest(tracer opentracing.Tracer) kithttp.RequestFunc {
+	return func(ctx context.Context, req *http.Request) context.Context {
+		// Try to find a Span in the Context.
+		if span := opentracing.SpanFromContext(ctx); span != nil {
+			// There's nothing we can do with any errors here.
+			_ = tracer.Inject(span, opentracing.TextMap, opentracing.HTTPHeaderTextMapCarrier(req.Header))
```

Comment:
useful to capture `http.url` and peer tags (unless can be done in middleware funcs)